### PR TITLE
Parquet WAL Improvements: Multitenant envs

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
A number of performance improvements for the Parquet wal that are particularly impactful in a highly multitenant environment:

- shared writer schema
- 1KB page buffer size for writers
- don't load page index or bloom filters when opening a wal "page" file
- lazy load wal "page" files

The real winners here were the 1k page buffer size and the lazy loading of the wal page files (instead of loading them immediately after flushing).

By default the page buffer size is 256KB. This page buffer is applied to every column of every tenant which adds up fast. Ideally there would be a way to specify per column page buffers, but this does not exist. The 1KB size does slow down tests. It doubles the length of time to run the tests in the `/tempodb/wal` folder for me, but in practice it seemed to have negligible impact on flush times in real environments.

### CODE REVIEW

**Review by mdisibio — APPROVED** [review]
Nice improvement.  Left one question. Probably ok since tests are passing.

File: tempodb/encoding/vparquet/wal_block.go [hunk 1]
```diff
@@ -320,20 +348,18 @@ func (b *walBlock) Flush() (err error) {
 		return fmt.Errorf("error closing writer: %w", err)
 	}
 
-	err = b.file.Close()
+	info, err := b.file.Stat()
 	if err != nil {
-		return fmt.Errorf("error closing file: %w", err)
+		return fmt.Errorf("error getting info: %w", err)
 	}
+	sz := info.Size()
```
[the remark this answers is not in the document]
Comment (joe-elliott) [reply tempodb/encoding/vparquet/wal_block.go:355]: Like a race condition between `writer.Close()` and `file.Stat()`? I think that as long as parquet-go isn't doing something async in `writer.Close()` this is fine.

**Comment by joe-elliott** [comment]
This has been approved and tests are passing, but I'm holding on merging. Still investigating the behavior of some small changes.

**Comment by joe-elliott** [comment]
Currently waiting on this PR: https://github.com/grafana/tempo/pull/1953 to prevent a performance regression in this branch.

**Pull request merged**